### PR TITLE
Storage metrics

### PR DESCRIPTION
**Pull request by iri989**

### CODE REVIEW

**Comment by eliasgomes** [comment]
Please rebase this pull request with `master`.

File: cosmic-metrics-collector/src/main/java/com/github/missioncriticalcloud/cosmic/metricscollector/tasks/StorageMetricCollector.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    ) {
+        this.storageMetricsRepository = storageMetricsRepository;
+        this.amqpTemplate = amqpTemplate;
+        this.metricWriter = metricWriter;
+        this.brokerExchange = brokerExchange;
+        this.brokerExchangeKey = brokerExchangeKey;
+    }
+
+    @Override
+    @Scheduled(cron = "${cosmic.metrics-collector.scan-interval:'0 */15 * * * *'}")
```
Comment (borisroman) [inline cosmic-metrics-collector/src/main/java/com/github/missioncriticalcloud/cosmic/metricscollector/tasks/StorageMetricCollector.java]: Don't we want to define the interval a bit more central?

File: cosmic-metrics-collector/src/test/resources/test-schema.sql [hunk 2]
```diff
@@ -130,3 +130,58 @@ CREATE TABLE IF NOT EXISTS `service_offering` (
...
+  KEY `i_volumes__data_center_id` (`data_center_id`),
+  KEY `i_volumes__account_id` (`account_id`),
+  KEY `i_volumes__pool_id` (`pool_id`),
+  KEY `i_volumes__last_pool_id` (`last_pool_id`),
+  KEY `i_volumes__instance_id` (`instance_id`),
+  KEY `i_volumes__state` (`state`),
+  KEY `i_volumes__update_count` (`update_count`)
+--   CONSTRAINT `fk_volumes__instance_id` FOREIGN KEY (`instance_id`) REFERENCES `vm_instance` (`id`) ON DELETE CASCADE,
+--   CONSTRAINT `fk_volumes__pool_id` FOREIGN KEY (`pool_id`) REFERENCES `storage_pool` (`id`)
+) ENGINE=InnoDB AUTO_INCREMENT=6 DEFAULT CHARSET=utf8;
```
Comment (borisroman) [inline cosmic-metrics-collector/src/test/resources/test-schema.sql:236]: New line :)

File: cosmic-metrics-collector/src/main/java/com/github/missioncriticalcloud/cosmic/metricscollector/repositories/jdbc/StorageMetricsJdbcRepository.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+public class StorageMetricsJdbcRepository implements StorageMetricsRepository{
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+    private final Properties queries;
+
+    @Autowired
+    public StorageMetricsJdbcRepository(final NamedParameterJdbcTemplate jdbcTemplate, @Qualifier("queries") final Properties queries) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.queries = queries;
+    }
+
```
Comment (borisroman) [inline cosmic-metrics-collector/src/main/java/com/github/missioncriticalcloud/cosmic/metricscollector/repositories/jdbc/StorageMetricsJdbcRepository.java:25]: Abundant white spaces.

File: cosmic-metrics-collector/src/test/java/com/github/missioncriticalcloud/cosmic/metricscollector/repositories/StorageMetricsJdbcRepositoryIT.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.*;
```
Comment (borisroman) [inline cosmic-metrics-collector/src/test/java/com/github/missioncriticalcloud/cosmic/metricscollector/repositories/StorageMetricsJdbcRepositoryIT.java]: Don't do wildcard imports

File: cosmic-metrics-collector/src/test/resources/test-data.sql [hunk 5]
```diff
@@ -17,3 +17,8 @@ INSERT INTO `service_offering`
   (id, cpu, ram_size)
 VALUES
   (1, 2, 4096);
+
+INSERT INTO `volumes`
+  (id, account_id, domain_id, size, data_center_id, volume_type, disk_offering_id, uuid)
+VALUES
+  (1, 1, 1, 840105984, 1, 'ROOT',	11, 'storage_uuid1');
```
Comment (borisroman) [inline cosmic-metrics-collector/src/test/resources/test-data.sql]: Spacing before `11`

**Review by borisroman — DISMISSED** [review]

File: cosmic-metrics-collector/src/main/java/com/github/missioncriticalcloud/cosmic/metricscollector/repositories/jdbc/StorageMetricsJdbcRepository.java (hunk 3 above)
Comment (bheuvel) [reply cosmic-metrics-collector/src/main/java/com/github/missioncriticalcloud/cosmic/metricscollector/repositories/jdbc/StorageMetricsJdbcRepository.java:25]: white spaces => white lines

**Comment by eliasgomes** [comment]
@borisroman The changes you requested were done. I'm dismissing your review so we can merge this PR.

**Pull request merged**